tests/ha_master/run.sh

```diff
@@ -12,6 +12,171 @@ MASTER_PORT3=8461
 MASTER_PORT4=8561
 MASTER_PORT5=8661
 
+function test_list_member() {
+    echo "[$(date)] <<<<<< start test_list_member_command >>>>>>"
+
+    master_ports=(0 $MASTER_PORT1 $MASTER_PORT2 $MASTER_PORT3 $MASTER_PORT4 $MASTER_PORT5)
+
+    alive=(1 2 3 4 5)
+    leaders=()
+
+    # get leader in all masters
+    for idx in ${alive[@]}; do
+        leaders+=($(get_leader $WORK_DIR 127.0.0.1:${master_ports[$idx]}))
+    done
+
+    # check leader is same for every master
+    leader=${leaders[0]}
+    leader_idx=${leader:6}
+    echo "current leader is" $leader
+    for ld in ${leaders[@]}; do
+        if [ "$leader" != "$ld" ]; then
+            echo "leader not consisent"
+            exit 1
+        fi
+    done
+
+    # check list-member master
+    run_dm_ctl_with_retry $WORK_DIR "127.0.0.1:$MASTER_PORT" \
+        "list-member --type=master" \
+        "\"alive\": true" 5
+
+    # kill leader
+    echo "kill leader" $leader
+    ps aux | grep $leader |awk '{print $2}'|xargs kill || true
+    check_port_offline ${master_ports[$leader_idx]} 20
+    sleep 5
+
+    # test again
+    alive=( "${alive[@]/$leader_idx}" )
+    leaders=()
+    for idx in ${alive[@]}; do
+        leaders+=($(get_leader $WORK_DIR 127.0.0.1:${master_ports[$idx]}))
+    done
+    leader=${leaders[0]}
+    leader_idx=${leader:6}
+    echo "current leader is" $leader
+    for ld in ${leaders[@]}; do
+        if [ "$leader" != "$ld" ]; then
+            echo "leader not consisent"
+            exit 1
+        fi
+    done
+    run_dm_ctl_with_retry $WORK_DIR "127.0.0.1:$MASTER_PORT" \
+        "list-member --type=master" \
+        "\"alive\": true" 4
+    echo "kill leader" $leader
+    ps aux | grep $leader |awk '{print $2}'|xargs kill || true
+    check_port_offline ${master_ports[$leader_idx]} 20
+    sleep 5
```

Comment:
It sames that these code are exactly the same with L73-94. Can we put them in one for-loop?